service/src/main/java/uk/gov/hmcts/reform/fpl/service/CaseSubmissionValidatorService.java

```diff
@@ -0,0 +1,57 @@
+package uk.gov.hmcts.reform.fpl.service;
+
+import com.google.common.collect.ImmutableList;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.fpl.enums.SectionType;
+import uk.gov.hmcts.reform.fpl.model.CaseData;
+
+import java.util.Collection;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+import javax.validation.ConstraintViolation;
+import javax.validation.Validation;
+import javax.validation.Validator;
+
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.APPLICANT;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.CASENAME;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.CHILDREN;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.DOCUMENTS;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.GROUNDS;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.HEARING;
+import static uk.gov.hmcts.reform.fpl.enums.SectionType.ORDERS;
+
+@Service
+public class CaseSubmissionValidatorService {
+
+    public List<String> validateCaseDetails(CaseData caseData) {
+        ImmutableList.Builder<String> caseErrors = ImmutableList.builder();
+
+        Validator validator = Validation.buildDefaultValidatorFactory().getValidator();
+        Set<ConstraintViolation<CaseData>> violations = validator.validate(caseData);
+
+        Stream.of(APPLICANT, CHILDREN, ORDERS, GROUNDS, HEARING, DOCUMENTS, CASENAME)
+            .flatMap(section -> Stream.of(groupErrorsBySection(violations, section)))
+            .flatMap(Collection::stream)
+            .forEach(caseErrors::add);
```

Comment:
Bit wired way, wouldn't it be better to just map main stream to collection of strings.

Reply:
No because the path JSR associates with a violation is not always the same as section name. e.g in case data orders and directions is labelled 'orders' while I need the section header (group) to read 'Orders and directions needed'.

Reply:
At the time you do `.forEach(caseErrors::add);` you have everything in almost final format. Doing another transformation seems more natural to me than iterating and pushing to another list.